pkg/sanity/sanity.go

```diff
@@ -55,6 +55,13 @@ type Config struct {
 	TestVolumeSize           int64
 	TestVolumeParametersFile string
 	TestVolumeParameters     map[string]string
+
+	// Callback functions to customize the creation of target and staging
+	// directories. Returns the new paths for mount and staging.
+	// If not defined, directories are created in the default way at TargetPath
+	// and StagingPath.
+	CreateTargetDir  func(path string) (string, error)
+	CreateStagingDir func(path string) (string, error)
```

Comment:
I'm fine with not exposing this functionality in the command line tool.

If someone needs it, we could add it by letting the test command invoke an external tool.